frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -86,19 +89,28 @@ public boolean hasNoWorker(String modelName) {
         if (currentWorkers < minWorker) {
             return addThreads(threads, model, minWorker - currentWorkers);
         } else {
-            for (int i = currentWorkers - 1; i >= minWorker; --i) {
-                WorkerThread thread = threads.remove(i);
-                thread.shutdown();
+            for (int i = 0; i < currentWorkers - 1; i++) {
```

Comment:
You have to iterate backward. otherwise, threads.remove(i) will failed with ArrayIndexOutofBoundException, or ConcurrentModificationException

Reply:
Should we do two pass scan?
1st scan for all workers in bad state, and then blindly kill workers.